src/skvhttpclient/test/ClientIntegrationTest.cpp

```diff
@@ -350,6 +363,101 @@ void testQuery1() {
 
 }
 
+void testPartialUpdate() {
+    auto&& [status, schemaPtr] = client.getSchema(collectionName, schemaName, 1).get();
+    K2EXPECT(k2::log::httpclient, status.is2xxOK(), true);
+
+    auto&& [beginStatus, txn] = client.beginTxn(dto::TxnOptions{}).get();
+    K2EXPECT(k2::log::httpclient, beginStatus.is2xxOK(), true);
+    auto record = buildRecord(collectionName, schemaPtr, std::string("A1"), std::string("B1"), 33, std::string("Test1"));
+    {
+      auto&& [writeStatus] = txn.write(record).get();
+      K2EXPECT(k2::log::httpclient, writeStatus.is2xxOK(), true);
+    }
+    auto key = buildRecord(collectionName, schemaPtr, std::string("A1"), std::string("B1"));
+    {
+        auto&& [readStatus, readRecord] = txn.read(key).get();
+        K2EXPECT(k2::log::httpclient, readStatus.is2xxOK(), true);
+        verifyEqual(readRecord, record);
+    }
+    {
+        auto record1 = buildRecord(collectionName, schemaPtr, std::string("A1"), std::string("B1"), 34, std::string("Test1_Update"));
+        std::vector<uint32_t> fields = {3}; // Update data field only
+        auto&& [writeStatus] = txn.partialUpdate(record1, fields).get();
+        K2EXPECT(k2::log::httpclient, writeStatus.is2xxOK(), true);
+        auto&& [readStatus, readRecord] = txn.read(key).get();
+        K2EXPECT(k2::log::httpclient, readStatus.is2xxOK(), true);
+        auto compareRecord = buildRecord(collectionName, schemaPtr, std::string("A1"), std::string("B1"), 33, std::string("Test1_Update"));
+        verifyEqual(readRecord, compareRecord);
+    }
+    {
+        K2LOG_I(k2::log::httpclient, "Full write, should succeed");
+        // Full write updating balance and data, should succeed
+        auto record1 = buildRecord(collectionName, schemaPtr, std::string("A1"), std::string("B1"), 35, std::string("Test1_Update1"));
+        auto key = buildRecord(collectionName, schemaPtr, std::string("A1"), std::string("B1"));
+
+        auto&& [writeStatus] = txn.write(record1).get();
+        K2EXPECT(k2::log::httpclient, writeStatus.is2xxOK(), true);
+        auto&& [readStatus, readRecord] = txn.read(key).get();
+        K2EXPECT(k2::log::httpclient, readStatus.is2xxOK(), true);
+        verifyEqual(readRecord, record1); // Not working, expected 35 getting 33
```

Comment:
This is interesting, I will take a look using your branch

Reply:
It's working now. Thanks for the fix!

Reply:
Great! Can you remove the comment so we don't get confused in the future?

Reply:
Done.